Merge add-controllers

IIExpanse opened the pull request:
Андрей, добрый день.
В соответствии с требованиями ТЗ и тестов реализован функционал для добавления, получения и обновления вещей, а также для добавления, получения, обновления и удаления пользователей.
Директории, функционал которых нужно реализовывать в будущих спринтах (request, booking), оставлены без изменений.

IIExpanse committed:
feat: Реализован функционал для добавления, получения и обновления вещей, а также для добавления, получения, обновления и удаления пользователей.

andrew-larkin commented on src/main/java/ru/practicum/shareit/item/controller/ItemController.java:
```diff
@@ -0,0 +1,157 @@
+@RestController
+@RequestMapping(path = "/items")
+@Slf4j
+@AllArgsConstructor
```
возможно, здесь лучше подойдет @RequiredArgsConstructor - он добавит в конструктор только final переменные, которые мы хотим внедрить в виде зависимости и не зааффектит, если мы добавим не final переменные в дальнейшем

andrew-larkin commented on src/main/java/ru/practicum/shareit/item/mapper/ItemMapper.java:
```diff
@@ -0,0 +1,25 @@
+import ru.practicum.shareit.item.dto.ItemDto;
+import ru.practicum.shareit.item.model.Item;
+
+public class ItemMapper {
```
если есть время, предлагаю взглянуть на маленькую библиотеку mapStruct, она автоматизирует создание мапперов, код написанный ниже превратится в одну строчку. Но это только на твое усмотрение

andrew-larkin commented on src/main/java/ru/practicum/shareit/user/controller/UserController.java:
```diff
@@ -0,0 +1,136 @@
+@RestController
+@RequestMapping(path = "/users")
+@Slf4j
+@AllArgsConstructor
```
@RequiredArgsConstructor

andrew-larkin commented on src/test/java/ru/practicum/shareit/item/repository/InMemoryItemRepositoryTest.java:
```diff
@@ -0,0 +1,167 @@
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class InMemoryItemRepositoryTest {
+
+    ItemRepository itemRepository;
```
в тестах переменные лучше указывать с модификатором доступа private

andrew-larkin commented on src/test/java/ru/practicum/shareit/user/controller/UserControllerTest.java:
```diff
@@ -0,0 +1,247 @@
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_EACH_TEST_METHOD)
+public class UserControllerTest {
```
также если будет время советую посмотреть на mockMvc. его используют почаще, чем testRestTemplate, он в дальнейшем очень пригодится. Но это также - только по твоему желанию. Так тесты написаны хорошо

andrew-larkin commented on src/test/java/ru/practicum/shareit/user/controller/UserControllerTest.java:
```diff
@@ -0,0 +1,247 @@
+
+    @Test
+    public void getUserTest() {
+        UserDto userDto = makeDefaultUserDto();
```
чтобы не дублировать создание dto в каждом тесте, предлагаю вынести создание в отдельный приватный метод, в котором будет создаваться "корректный" dto. Затем ты будешь вызывать этот метод в тестах, и там где нужно изменить какое-то поле dto, просто воспользуешься сеттером

andrew-larkin reviewed (changes requested):
Привет! Хочу сказать, что код очень качественный, промышленного уровня. Логика, конструкции реализованы очень удачно. Остается немного отрефакторить и будет готово. Если будут вопросы по правкам - можно писать мне в Слак.

IIExpanse committed:
fix: 1. Для контроллеров используется @RequiredArgsConstructor.
2. В полях тестов используются модификаторы private.
3. Тесты контроллеров переделаны под использование MockMvc вместо TestRestTemplate.
4. Значения DTO-объектов в тестах теперь задаются через сеттеры после получения шаблонного объекта.

feat: Мапперы создаются через mapStruct. Добавлена соответствующая зависимость.

Nagash40 commented:
Привет, отправляю на ревью новый коммит.
Правки добавлены по всем комментариям.
